lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2953,6 +2953,18 @@ namespace {
       if (decl->isVariadic()) {
         Impl.markUnavailable(result, "Variadic function is unavailable");
       }
+
+      if (decl->hasAttr<clang::ReturnsTwiceAttr>()) {
+        // The Clang 'returns_twice' attribute is used for functions like
+        // 'vfork' or 'setjmp'. Because these functions may return control flow
+        // of a Swift program to an arbitrary point, Swift's guarantees of
+        // definitive initialization of variables cannot be upheld. As a result,
+        // functions like these cannot be used in Swift.
+        Impl.markUnavailable(
+          result,
+          "Function that may return more than one time (annotated with the "
+          "'returns_twice' attribute) is unavailable");
```

Comment:
Nitpick: despite the variadic message above, I think it's better to phrase this as a general statement, i.e. *"functions* that may…are unavailable". That makes it clearer that it's not just a one-off rule. It may also be worth adding "in Swift", since that's an important part of it.